graylog2-web-interface/src/views/components/aggregationbuilder/AggregationBuilder.jsx

```diff
@@ -37,12 +37,19 @@ export type VisualizationComponentProps = {|
   onChange: OnVisualizationConfigChange,
   width: number,
   toggleEdit: () => void,
-|};
+};
 
 export type VisualizationComponent =
-  { type?: string, propTypes?: any }
+  { type: string, propTypes?: any }
   & React.ComponentType<VisualizationComponentProps>;
 
+export const makeVisualization = (component: React.ComponentType<VisualizationComponentProps>, type: string): VisualizationComponent => {
```

Comment:
Due to the existing prop type warning we are receiving for some visualizations, I found out the component name gets lost here. Currently the warning looks like:
```
Warning: Failed prop type: Invalid prop `data` of type `object` supplied to `<<anonymous>>`, expected an array.
```

Which means we need to add something like `LineVisualization.displayName = 'LineVisualization'` to every visualization, or we add an argument to this function `makeVisualization(component, displayName, type)` and add it here with `visualizationComponent.displayName = displayName`.

The first solution looks more familiar because the `displayName` clearly belongs to the component, but I prefer the second approach, because we only need to add the `displayName` when we use `makeVisualization()` and flow will remind us, whenever we could forget about adding the attribute.

Reply:
I think this happens for the `BarVisualization` component? In that case, I think the display name never got lost, because it was simply never present as it is a functional component. We could use the logic from `connect.jsx`:

```
  const wrappedComponentName = Component.displayName || Component.name || 'Unknown/Anonymous';
```
and use that as `displayName`, but we would not gain anything.

Long story short: I think we should just set `displayName` explicitly, when creating a functional component. E.g. adding: `BarVisualization.displayName = 'BarVisualization';` before exporting it. This is beyond the scope of this PR though.

Reply:
You are right, it never gets lost, I thought so because the components name was part of the warning before these changes.
I agree adding the `displayName` for functional components is a good idea in general, but out of the scope. Looks like a goos case for an ESLint rule, i will have a look at a possible solution.